### ISSUE TITLE
custom transformer for `Bun.Transpiler`

### ISSUE THREAD

**Issue by HairyRabbit** [oven-sh/bun#3936]
### What is the problem this feature would solve?

would we plan to support custom transformers ?

### What is the feature you are proposing to solve the problem?

like tsc transpiler:

```ts
ts.transpileModule(source, {
  //...
  transformers: {
    before: [],
    after: [],
  }
})
```

**Comment by Electroid** [comment]
Are you referring to AST transformers?

**Comment by HairyRabbit** [comment]
yup, i need rewrite ast before build.

**Comment by Electroid** [comment]
Closing as duplicate of #2729
